Lib/test/test_codecs.py

```diff
@@ -712,9 +712,8 @@ def test_bug691291(self):
         self.addCleanup(support.unlink, support.TESTFN)
         with open(support.TESTFN, 'wb') as fp:
             fp.write(s)
-        with support.check_warnings(('', DeprecationWarning)):
-            reader = codecs.open(support.TESTFN, 'U', encoding=self.encoding)
-        with reader:
+        with codecs.open(support.TESTFN, 'r',
```

Comment:
"Binary mode" has different meaning in Python 2 than in Python 3, and this test refers to the former meaning. I think this test can be removed now. The code which it tests was removed years ago.

But you can add new test for modes `'U'`, `'rU'`, `'r+U'`, `'rt'`, `'wt'`, `'at'`, `'r+t'` which work with builtin `open()` in Python 2, but are not supported by `codecs.open()`.

Reply:
> I think this test can be removed now. The code which it tests was removed years ago.

That sounds to be unrelated to https://bugs.python.org/issue37330 : if you would like to remove the test, I would prefer that you submit a separated PR.

> But you can add new test for modes 'U', 'rU', 'r+U'

Done

> ...  'rt', 'wt', 'at', 'r+t' which work with builtin open() in Python 2, but are not supported by codecs.open().

These modes are accepted by codecs.open().

Reply:
I get an error.
```
>>> import codecs
>>> codecs.open('/dev/null', 'rt', encoding='utf-8')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/serhiy/py/cpython/Lib/codecs.py", line 905, in open
    file = builtins.open(filename, mode, buffering)
ValueError: can't have text and binary mode at once
```